frontend/event/eventController.js

```diff
@@ -79,9 +90,117 @@
+         * @param {object} event - The current event
+         */
+        eventCtrl.getProfileColor = (event) => {
+            const profile = _.filter(eventCtrl.user.institution_profiles, function(prof) {
+                return prof.institution_key === event.institution_key;
+            });
+            if(profile.length > 0) {
+                return profile[0].color;
+            }
+            return 'teal';
+        };
+
+        /**
+         * Loads the events when the filters of month and/or year is changed
+         */
+        eventCtrl.loadFilteredEvents = () => {
+            moreEvents = true;
+            actualPage = 0;
+            isAnotherMonth = true;
+            eventCtrl.loadMoreEvents();
+        };
+
+        /**
+         * Distributes events on days that happens 
+         * @param {object} event the event that be distributed
+         * @param {object} eventsByDay the object with days in keys
+         */
+        eventCtrl._distributeEvents = (event, eventsByDay) => {
+            const daysOfCurrentMonth = new Date(eventCtrl.currentYear, eventCtrl.currentMonth.month, 0).getDate();
+            const startMonth = new Date(event.start_time).getMonth()+1;
+            const endMonth = new Date(event.end_time).getMonth()+1;
+            let day = new Date(event.start_time).getDate();
+            let endDay = new Date(event.end_time).getDate();
+            if(endDay <= day && eventCtrl.currentMonth.month < endMonth) {
+                endDay = daysOfCurrentMonth;
+            } else if(endDay <= day && startMonth < endMonth
+                && eventCtrl.currentMonth.month === endMonth) {
+                day = 1;
+            }
+            for (let i = day; i <= endDay; i++) {
+                if(startMonth === eventCtrl.currentMonth.month
+                    || endMonth === eventCtrl.currentMonth.month) {
+                    if(!eventsByDay[i]) 
+                        eventsByDay[i] = [];
+                    eventsByDay[i].push(event);
+                }
+            }
+        };
+
+        /**
+         * Group the events into an array of days of the selected month of year
+         * @private
+         */
+        eventCtrl._getEventsByDay = () => {
+            if(eventCtrl.events.length > 0) {
+                eventCtrl.eventsByDay = [];
+                let eventsByDay = {};
+                _.forEach(eventCtrl.events, function(event) {
+                    eventCtrl._distributeEvents(event, eventsByDay);
+                });
+
+                let days = Object.keys(eventsByDay);
+                _.forEach(days, function(day) {
+                    let currentValue = {
+                        day: day,
+                        events: eventsByDay[day]
+                    };
+                    eventCtrl.eventsByDay.push(currentValue);
+                });
+            }
+        };
+
+        /**
+         * Loads the years 2017 to 2050 to show in filter by year
```

Comment:
get the current year, add some constant quantity and set the limit to that year, it's better.